data/modules/FlightLog/FlightLogEntries.lua

```diff
@@ -0,0 +1,397 @@
+-- Copyright © 2008-2023 Pioneer Developers. See AUTHORS.txt for details
+-- Licensed under the terms of the GPL v3. See licenses/GPL-3.txt
+
+-- Entries for the FlightLog
+
+local Game = require 'Game'
+local Format = require 'Format'
+
+local utils = require 'utils'
+
+local Character = require 'Character'
+
+-- required for formatting / localisation
+local ui = require 'pigui'
+local Lang = require 'Lang'
+local l = Lang.GetResource("ui-core")
+-- end of formating / localisation stuff
+
+FlightLogEntry = {}
+
+-- how many default (so not custom) elements do we have
+---@type integer
+FlightLogEntry.TotalDefaultElements = 0
+
+
+---@class FlightLogEntry.Base
+---A generic log entry:
+---@field protected sort_date number Seconds since the epoch in game time, used to sort when loading old style
+---@field protected entry string User entered text associated with the entry
+---@field protected always_custom boolean|nil Is this always treated as a custom entry (so not auto deleted)
+FlightLogEntry.Base = utils.class("FlightLogEntry")
+
+---@return string Description of this type
+function FlightLogEntry.Base:GetType()
+	local full_type = self.Class().meta.class
+	return string.sub( full_type, #"FlightLogEntry."+1, #full_type )
+end
```

Comment:
Yes, I really didn't like this either; I was hoping to find some way to 'automagically' solve the problem.

In the end I've gone for overloaded methods each returning a constant, only because I'm uncomfortable with my inability to make a variable const and this makes it harder (although far from impossible) to accidentally break.

Reply:
That's a good argument for making this a function (in addition to the one of maintaining a consistent/uniform API) - I find in practice overwriting class-level variables isn't a significant concern especially with static analysis assistance from a language server, but it's an insignificant performance overhead so safer is better as long as productivity doesn't suffer.